pkg/languagedetection/privileged_detector.go

```diff
@@ -53,3 +93,31 @@ func MockPrivilegedDetectors(t *testing.T, newDetectors []languagemodels.Detecto
 	t.Cleanup(func() { detectorsWithPrivilege = oldDetectors })
 	detectorsWithPrivilege = newDetectors
 }
+
+func (l PrivilegedLanguageDetector) getBinID(process languagemodels.Process) (binaryID, error) {
+	procPath := filepath.Join(l.hostProc, strconv.Itoa(int(process.GetPid())))
+	exePath := filepath.Join(procPath, "exe")
+	binPath, err := os.Readlink(exePath)
+	if err != nil {
+		return binaryID{}, fmt.Errorf("readlink %s: %v", exePath, err)
+	}
+
+	binPath = filepath.Join(procPath, "root", binPath)
+
+	var stat syscall.Stat_t
+	err = syscall.Stat(binPath, &stat)
+	if err != nil {
+		return binaryID{}, fmt.Errorf("stat binary path %s: %v", binPath, err)
+	}
+
+	return binaryID{
+		major: unix.Major(stat.Dev),
+		minor: unix.Minor(stat.Dev),
```

Comment:
Any reason to separate major and minor here?

Reply:
I was mostly copying what @guyarb did in his PR https://github.com/DataDog/datadog-agent/pull/18420/files#diff-37e2d61231cb67aefea0005caebd183eecfbd77d1263f40ddfe90267f57aa740R356-L349. Looking at it again, I suspect that this is because they are used separately. I will recombine the two